api/auth/auth.go

```diff
@@ -0,0 +1,211 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ava-labs/gecko/utils/timer"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+const (
+	headerKey      = "Authorization"
+	headerValStart = "Bearer "
+	// Endpoint is the base of the auth URL
+	Endpoint       = "auth"
+	maxPasswordLen = 1024
+)
+
+var (
+	// TokenLifespan is how long a token lives before it expires
+	TokenLifespan = time.Hour * 12
+
+	// ErrNoToken is returned by GetToken if no token is provided
+	ErrNoToken = errors.New("auth token not provided")
+
+	errWrongPassword = errors.New("incorrect password")
+)
+
+// Auth handles HTTP API authorization for this node
+type Auth struct {
+	lock     sync.RWMutex // Prevent race condition when accessing password
+	Enabled  bool         // True iff API calls need auth token
```

Comment:
typo `iff`

Reply:
`iff` is `if and only if`, though admittedly this may be too jargony

Reply:
Ah, I'll admit I didn't pick up on it. But glad I know it now.